lib/jxl/epf.cc

```diff
@@ -648,13 +657,24 @@ FilterPipeline* PrepareFilterPipeline(
     const Rect& input_rect, size_t image_ysize, size_t thread,
     Image3F* JXL_RESTRICT out, const Rect& output_rect) {
   const LoopFilter& lf = dec_state->shared->frame_header.loop_filter;
-  JXL_DASSERT(image_rect.x0() % GroupBorderAssigner::kPaddingXRound == 0);
-  JXL_DASSERT(input_rect.x0() % GroupBorderAssigner::kPaddingXRound == 0);
-  JXL_DASSERT(output_rect.x0() % GroupBorderAssigner::kPaddingXRound == 0);
-  JXL_DASSERT(input_rect.x0() >= lf.Padding());
+  // image_rect, input and output must all have the same kPaddingXRound
+  // alignment for SIMD, but it doesn't need to be 0.
+  JXL_DASSERT(image_rect.x0() % GroupBorderAssigner::kPaddingXRound ==
```

Comment:
I think we actually need it to be 0 because we do aligned loads

Reply:
The filter function will get a different x0 because it needs to process border needed for future stages; instead of `filter_x0 = RoundDown(RoundDown(x0, 4) - border, 4)` this patch does `filter_x0 = RoundDown(x0 - border, 4)` so the image_rect.x0 here doesn't need to aligned.

Reply:
Ah, I didn't notice this was epf.cc :) nvm...